geowebcache/diskquota/jdbc/src/main/java/org/geowebcache/diskquota/jdbc/JDBCConfiguration.java

```diff
@@ -369,4 +371,34 @@ public String toString() {
 
     }
 
+    /**
+     * 
+     * 
+     * @param allowEnvParametrization
+     * @return
+     */
+    public JDBCConfiguration clone(boolean allowEnvParametrization) {
+        
+        JDBCConfiguration conf = (JDBCConfiguration) SerializationUtils.clone(this);
+        
+        final GeoWebCacheEnvironment gwcEnvironment = GeoWebCacheExtensions.bean(GeoWebCacheEnvironment.class);
+        
+        if (allowEnvParametrization &&  gwcEnvironment != null && GeoWebCacheEnvironment.ALLOW_ENV_PARAMETRIZATION) {
+            conf.setDialect((String) gwcEnvironment.resolveValue(getDialect()));
+            conf.setJNDISource((String) gwcEnvironment.resolveValue(getJNDISource()));
+            ConnectionPoolConfiguration connectionPoolConfig = getConnectionPool();
+            if (connectionPoolConfig != null) {
+                ConnectionPoolConfiguration expConnectionPoolConfig = (ConnectionPoolConfiguration) SerializationUtils.clone(connectionPoolConfig);
+                expConnectionPoolConfig.setDriver((String) gwcEnvironment.resolveValue(connectionPoolConfig.getDriver()));
+                expConnectionPoolConfig.setUrl((String) gwcEnvironment.resolveValue(connectionPoolConfig.getUrl()));
+                expConnectionPoolConfig.setUsername((String) gwcEnvironment.resolveValue(connectionPoolConfig.getUsername()));
+                expConnectionPoolConfig.setPassword((String) gwcEnvironment.resolveValue(connectionPoolConfig.getPassword()));
+                expConnectionPoolConfig.setValidationQuery((String) gwcEnvironment.resolveValue(connectionPoolConfig.getValidationQuery()));
+                
```

Comment:
NOTICE that - JDBCConfiguration does not have PORT as configuration parameter. Everything is handled through the JDBC URL.